src/pyclaw/sharpclaw/solver.py

```diff
@@ -652,10 +655,12 @@ def get_dt_new(self):
                 # Step-size update of starting methods
                 sspcoeff_ratio = self.sspcoeff0/self.sspcoeff
                 self.dt = sspcoeff_ratio * self.dt * self.cfl_desired / cfl
-                if self.time_integrator == 'SSPLMMk3' and self.check_lmm_cond:
+                if self.time_integrator == 'SSPLMMk3' and self.check_lmm_cond and not self.lmm_cond:
```

Comment:
If `check_lmm_cond ` is `True`, then the step size conditions for 3rd order methods are being checked, otherwise not.
`lmm_cond` is `True (False)` if the step size conditions are satisfied (violated).

`lmm_cond` can be changed to `accept_lmm_cond`if this makes it clearer.
Also we can decide to always check these conditions and drop `check_lmm_cond`.